R/format.R

```diff
@@ -103,6 +103,21 @@ formatSignif = function(
 #'   \code{params = list('ko-KR', list(year = 'numeric', month = 'long', day =
 #'   'numeric'))}
 formatDate = function(table, columns, method = 'toDateString', params = NULL) {
+  x = table$x
+  if (x$filter != 'none') {
+    if (inherits(columns, 'formula')) columns = all.vars(columns)
+    colnames = base::attr(x, 'colnames', exact = TRUE)
+    rownames = base::attr(x, 'rownames', exact = TRUE)
+    if (is.null(params)) params = list()
+    cols = as.character(name2int(columns, colnames, rownames))
+    if (is.null(x$options$filterDateFmt)) x$options$filterDateFmt = list()
```

Comment:
I'd store the info in `x$filterDateFmt` instead of `x$options$filterDateFmt`, because `x$options` is for initializing a DataTable via `.DataTable()`, and `filterDateFmt` is not one of the supported options (which is why the current PR can work):
https://github.com/rstudio/DT/blob/8ba54abd4e4b1215908aae83981f4b51d3be1984/inst/htmlwidgets/datatables.js#L269

Accordingly, on the JS side, use `data.filterDateFmt` instead of `options.filterDateFmt`.